test/e2e/engine/template.go

```diff
@@ -178,6 +178,18 @@ func (e *Engine) HasTiller() bool {
 	return false
 }
 
+// TillerMaxHistory will return true if tiller addon is enabled
+func (e *Engine) TillerMaxHistory() string {
+	for _, addon := range e.ExpandedDefinition.Properties.OrchestratorProfile.KubernetesConfig.Addons {
+		if addon.Name == "tiller" {
+			if addon.Config != nil {
+				return addon.Config["max-history"]
+			}
+		}
+	}
+	return "5"
```

Comment:
Why return 5 here? Why not "-1" or nil?

Reply:
Actually just realized this is the default max history. In that case it should be inside the for loop (return 5 if addon tiller was found but config is nil) and then return -1 is no tiller addon was found.